exec/java-exec/src/main/java/org/apache/drill/exec/store/mock/ExtendedMockBatchReader.java

```diff
@@ -140,7 +133,7 @@ public boolean next() {
       writer.save();
     }
 
-    return true;
+    return !loader.atLimit();
```

Comment:
The "EVF V2" automatically checks the limit, it that correct?
https://github.com/apache/drill/blob/282b42256a996729d4079b52176c2ab7068d2de6/exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/scan/v3/lifecycle/ReaderLifecycle.java#L251-L253

Reply:
That is correct. This particular test uses a subset of EVF: just the `ResultSetLoader`, which is responsible for enforcing the per-reader limit. In the case of this mock reader for testing, there is only one reader, so enforcing the per-reader limit is equivalent to enforcing the scan limit.